packages/project-types/src/package.ts

```diff
@@ -0,0 +1,19 @@
+export interface Package {
+    name: string;
+    sapux: boolean | string[];
+    sapuxLayer?: UI5FlexLayer;
+    main?: string;
+    cds?: object;
+    dependencies?: { [dependencyName: string]: string };
+    devDependencies?: { [dependencyName: string]: string };
+    optionalDependencies?: object;
+    scripts?: { [scriptName: string]: string };
+    ui5?: object;
+    remarkConfig?: object;
+    version?: string;
+}
+
+export const enum UI5FlexLayer {
```

Comment:
What is the advantage of using an enum instead of `type UI5FlexLayer = 'VENDRO' | 'CUSTOMER_BASE'`?

Reply:
* The enum values are meant to be opaque to the consumer, so they don't really care what `UI5FlexLayer.VENDOR` actually resolves to
* We can iterate through the value of an Enum if we want to use that in a validator, for example
* The enum name `UI5FlexLayer` acts as sort of a namespace compared to the unadorned string `VENDOR`
* The transpiled code is going to be slightly more verbose compared to string types. If performance is a problem, it's probably better to profile first to see if this transpilation is a problem. The other considerations probably have more weight

In the end, I think choosing one over the other comes down to coding style

Reply:
> In the end, I think choosing one over the other comes down to coding style

I agree with you here because I doubt that perfomance is an issue. I have started to use simple types if I know the strings are constant, I only use an enum if I know the value can be changed in the future. In this case, I doubt that the values change, so I would switch to the simple type BUT as you said, it is coding style, so no need to change, just look at my argument again, and then decide :)

Reply:
Given the fact that 
a) the value here is very very unlikely to be changed
b) `const enum` are more problematic than string literals 

I've decided to change this to

```
export type UI5FlexLayer = 'VENDOR' | 'CUSTOMER_BASE';
```